base/deprecated.jl

```diff
@@ -1003,6 +1003,20 @@ macro vectorize_2arg(S,f)
 end
 export @vectorize_1arg, @vectorize_2arg
 
+# deprecations for uses of old dot operators (.* etc) as objects, rather than
+# just calling them infix.
+for op in (:(!=), :≠, :+, :-, :*, :/, :÷, :%, :<, :(<=), :≤, :(==), :>, :>=, :≥, :\, :^)
+    dotop = Symbol('.', op)
+    # define as const dotop = (a,b) -> ...
+    # to work around syntax deprecation for dotop(a,b) = ...
+    @eval const $dotop = (a,b) -> begin
+        depwarn(string($(string(dotop)), " is no longer a function object; use broadcast(",$op,", ...) instead"),
```

Comment:
In the examples below, did you intend for the first two to succeed without throwing a depwarn (in contrast to the third)?
```julia
julia> .+(eye(2), eye(2))
2×2 Array{Float64,2}:
 2.0  0.0
 0.0  2.0

julia> (.+)(eye(2), eye(2))
2×2 Array{Float64,2}:
 2.0  0.0
 0.0  2.0

julia> foo = .+; foo(eye(2), eye(2))
WARNING: .+ is no longer a function object; use broadcast(+, ...) instead
 in depwarn(::String, ::Symbol) at ./deprecated.jl:64
 in (::Base.##1025#1026)(::Array{Float64,2}, ::Array{Float64,2}) at ./deprecated.jl:1013
 in eval(::Module, ::Any) at ./boot.jl:236
 in eval(::Module, ::Any) at /Users/sacha/pkg/julia/usr/lib/julia/sys.dylib:?
 in eval_user_input(::Any, ::Base.REPL.REPLBackend) at ./REPL.jl:66
 in macro expansion at ./REPL.jl:97 [inlined]
 in (::Base.REPL.##3#4{Base.REPL.REPLBackend})() at ./event.jl:66
while loading no file, in expression starting on line 0
2×2 Array{Float64,2}:
 2.0  0.0
 0.0  2.0
```